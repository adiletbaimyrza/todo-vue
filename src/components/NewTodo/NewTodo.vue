<template>
  <div :class="$style['bar']">
    <CoreIcon
      :src="require('../../assets/circle.svg')"
      :class="$style['bar__icon']"
    />
    <input
      v-model="newTodoContent"
      @keyup.enter="addNewTodo"
      :class="$style['bar__input']"
      type="text"
      placeholder="Ex: Buy groceries..."
    />
  </div>
</template>

<script>
import CoreIcon from '../CoreIcon/CoreIcon.vue'

export default {
  name: 'NewTodo',
  components: { CoreIcon },
  data() {
    return {
      newTodoContent: '',
    }
  },
  methods: {
    addNewTodo() {
      if (this.newTodoContent.trim() !== '') {
        const newTodo = {
          id: Date.now(),
          content: this.newTodoContent,
          isCompleted: false,
        }
        this.$store.commit('add', newTodo)
        this.newTodoContent = ''
      }
    },
  },
}
</script>

<style lang="scss" module>
.bar {
  display: flex;
  width: 600px;
  background-color: var(--card-background-color);
  padding: 1em 1.5em;
  border-radius: 0.4em;
  margin-bottom: 1.5em;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
  caret-color: var(--font-color-regular);

  &__icon {
    margin-right: 1em;
  }

  &__input {
    flex-grow: 1;
    outline: none;
    border: none;
    font-size: 1em;
    color: var(--font-color-regular);
    background-color: var(--card-background-color);

    &::placeholder {
      color: #b1b1b1;
    }
  }
}
</style>
