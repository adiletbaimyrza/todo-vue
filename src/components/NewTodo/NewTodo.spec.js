import { shallowMount } from '@vue/test-utils'
import NewTodo from './NewTodo.vue'

/* global jest describe beforeEach it expect */
describe('NewTodo.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(NewTodo)
  })

  it('renders correctly', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('fires addNewTodo method on keyup.enter', async () => {
    const addNewTodoMock = jest.fn()
    wrapper.vm.addNewTodo = addNewTodoMock
    const input = wrapper.find('input')
    await input.setValue('Buy groceries')
    await input.trigger('keyup.enter')
    expect(addNewTodoMock).toHaveBeenCalled()
  })

  it('renders placeholder text correctly', () => {
    const input = wrapper.find('input')
    expect(input.attributes('placeholder')).toBe('Ex: Buy groceries...')
  })
})
