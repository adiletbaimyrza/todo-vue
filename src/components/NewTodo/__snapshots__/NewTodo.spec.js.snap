// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodo.vue renders correctly 1`] = `
<div
  class="bar"
  data-testid="NewTodo"
>
  <img
    alt="Icon"
    class="icon bar__icon"
    data-testid="CoreIcon"
    height="24px"
    src=""
    width="24px"
  />
  <input
    class="bar__input"
    data-testid="Input"
    maxlength="151"
    placeholder="Enter a new task..."
    type="text"
  />
  <!--v-if-->
  <!--v-if-->
</div>
`;
